log4j-core-test/src/main/java/org/apache/logging/log4j/core/test/junit/LoggingResolvers.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.logging.log4j.core.test.junit;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.lang.reflect.Parameter;
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.Logger;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.appender.AbstractManager;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.plugins.di.Keys;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.api.extension.support.TypeBasedParameterResolver;
+import org.junit.platform.commons.support.ReflectionSupport;
+
+/**
+ * Annotates a test extension annotation to indicate that it sets up Log4j and can resolve parameters of various kinds.
+ * These supported types include {@link LoggerContext}, {@link Configuration}, configured {@link Appender} plugins
+ * with a {@linkplain org.apache.logging.log4j.plugins.Named name}, named {@link AbstractManager} instances, and
+ * named {@link Logger} instances.
+ */
+@Target(ElementType.ANNOTATION_TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Inherited
+@Documented
+@Log4jTest
+@ExtendWith(LoggerContextResolver.class)
+@ExtendWith(ConfigurationResolver.class)
+@ExtendWith(AppenderResolver.class)
+@ExtendWith(AppenderManagerResolver.class)
+@ExtendWith(LoggerResolver.class)
+public @interface LoggingResolvers {}
+
+/**
+ * Handles parameter resolution of a {@link LoggerContext}.
+ */
+class LoggerContextResolver extends TypeBasedParameterResolver<LoggerContext> {
```

Comment:
Since all these `ParameterResolver` extensions are package-private and only used by this particular file, I combined them all into one file. Less lines wasted on license headers 😂